addons/account/tests/test_account_move_reconcile.py

```diff
@@ -4438,3 +4384,179 @@ def test_reconcile_payment_custom_rate(self):
+                }),
+                (0, 0, {
+                    'factor_percent': 60,
+                    'repartition_type': 'tax',
+                    'use_in_tax_closing': False,
+                }),
+            ],
+        })
+
+        invoice = self.env['account.move'].create([{
+            'move_type': 'out_invoice',
+            'partner_id': self.partner_a.id,
+            'date': '2017-01-01',
+            'invoice_date': '2017-01-01',
+            'invoice_line_ids': [
+                Command.create({
+                    'product_id': self.product_a.id,
+                    'price_unit': 800.0,
+                    'tax_ids': [Command.set(tax.ids)],
+                    'analytic_distribution': analytic_distribution_a,
+                }),
+                Command.create({
+                    'product_id': self.product_b.id,
+                    'price_unit': 800.0,
+                    'tax_ids': [Command.set(tax.ids)],
+                    'analytic_distribution': analytic_distribution_b,
+                }),
+                Command.create({
+                    'product_id': self.product_b.id,
+                    'price_unit': 800.0,
+                    'tax_ids': [Command.set(tax.ids)],
+                    'analytic_distribution': analytic_distribution_b,
+                }),
+            ]
+        }])
+        invoice.action_post()
+
+        self.env['account.payment.register'].with_context(active_model='account.move', active_ids=invoice.ids).create({
+            'payment_date': invoice.date,
+        })._create_payments()
+
+        # Check the analytic distribution on the caba move lines.
+        # I.e. lines with different analytic distributions should not be grouped / mixed together.
+        caba_move = invoice.tax_cash_basis_created_move_ids
+        expected_caba_move_line_values = [
+            {
+                'account_id': self.cash_basis_base_account.id,
+                'debit': 0.0,
+                'credit': 800.0,
+                'analytic_distribution': analytic_distribution_a_serialized,
+            },
+            {
+                'account_id': self.cash_basis_base_account.id,
+                'debit': 800.0,
+                'credit': 0.0,
+                'analytic_distribution': analytic_distribution_a_serialized,
+            },
+            {
+                'account_id': self.cash_basis_base_account.id,
+                'debit': 0.0,
+                'credit': 1600.0,
+                'analytic_distribution': analytic_distribution_b_serialized,
+            },
+            {
+                'account_id': self.cash_basis_base_account.id,
+                'debit': 1600.0,
+                'credit': 0.0,
+                'analytic_distribution': analytic_distribution_b_serialized,
+            },
+            {
+                'account_id': self.tax_account_1.id,
+                'debit': 0.0,
+                'credit': 192.0,
+                'analytic_distribution': False,
```

Comment:
Why not for this one ? :thinking: Shouldn't it be split in two, each of the with an analytic distribution. I don't get why they are grouped like this.

Reply:
I added a comment explaining the expected grouping of the lines.
What do you think?

Or should the behavior be different?
I did not change when the analytic distribution gets applied to the journal items. and I think it was discussed at some point that we just want to take the analytic distribution from there. But it was a long time ago so I am not 100% sure anymore (and am ofc. open to rediscuss)

Reply:
Ok, it's clearer :) 
Just double-checking the functional side with a PO (I'm curious ^^)

Reply:
What you did is correct :) 

There is one more case, though, which works with your PR already but should be tested IMO: If the "analytic" field of the account.tax is True, the lines made from repartition lines with use_in_tax_closing=True also need to receive the analytic distribution. So, I'd say you can copy-paste your whole test into a new one, just changing that :) Of course, adapt the comments accordingly.

Reply:
I added a new copy-pasted test (only changed what is necessary).

I explicitly set the `analytic` field on the original test to `False` to clarify. Should I rather not?